Foundation/NSXMLDocument.swift

```diff
@@ -53,147 +53,273 @@ public enum NSXMLDocumentContentKind : UInt {
 	@discussion Note: if the application of a method would result in more than one element in the children array, an exception is thrown. Trying to add a document, namespace, attribute, or node with a parent also throws an exception. To add a node with a parent first detach or create a copy of it.
 */
 public class NSXMLDocument : NSXMLNode {
-    
+    private var _xmlDoc: _CFXMLDocPtr {
+        get {
+            return _CFXMLDocPtr(_xmlNode)
+        }
+    }
     /*!
         @method initWithXMLString:options:error:
         @abstract Returns a document created from either XML or HTML, if the HTMLTidy option is set. Parse errors are returned in <tt>error</tt>.
     */
-    public convenience init(XMLString string: String, options mask: Int) throws { NSUnimplemented() }
-    
+    public convenience init(XMLString string: String, options mask: Int) throws {
+        let data: NSData = string.withCString {
```

Comment:
In the meanwhile I'm changing it to use `dataUsingEncoding` from NSString, which seems to avoid `CFStringCreateExternalRepresentation` already.